omc3/scripts/bad_bpms_summary.py

```diff
@@ -0,0 +1,294 @@
+    """ Create a TfsDataFrame with all bad-bpms within selcted dates.
+
+    Args:
+        root (Path): Root path to the GUI output folder.
+        dates (Sequence[Path | str]): List of dates or glob patterns to collect bad-bpms from.
+        accel_glob (str): Accelerator name (glob for the sub-directories).
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with all bad-bpms within selcted dates.
+    
+    """
+    dfs = []
+    for date in dates:
+        date_dir = root / date
+        if date_dir.is_dir():
+            dfs.append(collect_date(date_dir, accel_glob))
+            continue
+
+        for date_dir in root.glob(date):
+            dfs.append(collect_date(date_dir, accel_glob))
+    return tfs.concat(dfs, axis="index", ignore_index=True) 
+
+
+def collect_date(date_dir: Path, accel_glob: str) -> tfs.TfsDataFrame:
+    """ Collect bad-bpms for a single date.
+    
+    Args:
+        date_dir (Path): Path to the date directory.
+        accel_glob (str): Accelerator name (glob for the sub-directories).
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with all bad-bpms for the date.
+    """
+    dfs: list[tfs.TfsDataFrame] = []
+
+    for accel_dir in date_dir.glob(accel_glob):
+        measurements_dir = accel_dir / MEASUREMENT_DIR
+        if not measurements_dir.is_dir():
+            continue
+
+        for measurement in measurements_dir.iterdir():
+            if not measurement.is_dir():
+                continue
+
+            df_collected = collect_measurement_dir(measurement)
+            df_collected.loc[:, ACCEL] = accel_dir.name
+            dfs.append(df_collected)
+
+    if not len(dfs):
+        return get_empty_df()
+
+    return tfs.concat(dfs, axis="index", ignore_index=True) 
+    
+
+def collect_measurement_dir(measurement_dir: Path) -> tfs.TfsDataFrame:
+    """ Collect bad-bpms for a single measurement directory.
+    
+    Args:
+        measurement_dir (Path): Path to the measurement directory.
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with all bad-bpms from the measurement directory.
+    
+    """
+    dfs: list[tfs.TfsDataFrame] = []
+
+    readers_map = {
+        BAD_BPMS_HARPY: read_harpy_bad_bpms_file,
+        BAD_BPMS_IFOREST: read_iforest_bad_bpms_file
+    }
+
+    for glob_pattern, reader in readers_map.items():
+        for bad_bpms_file in measurement_dir.glob(glob_pattern):
+            dfs.append(reader(bad_bpms_file))
+
+    if not len(dfs):
+        return get_empty_df()
+    
+    return tfs.concat(dfs, axis="index", ignore_index=True)
+
+
+# File Readers --
+            
+def read_harpy_bad_bpms_file(svd_file: Path) -> tfs.TfsDataFrame:
+    """ Reads a harpy bad-bpm file and returns a TfsDataFrame with all unique bad-bpms.
+    
+    Args:
+        svd_file (Path): Path to the bad-bpm file.
+
+    Returns:
+        tfs.TfsDataFrame: TfsDataFrame with all unique bad-bpms.
+
+    """
+    TO_INGNORE = ["not found in model", "known bad bpm"]
```

Comment:
but they are not bad because the analysis says they are bad. They are bad, because we say they are bad. It's a bit of a self-fullfilling prophecy then.... but I have an idea

Reply:
True, I guess it's not obvious here. Let's see your idea.